### PR TITLE
Fixed SelectedObjects bugs

### PR DESCRIPTION
**Pull request by iatealemon**
fixed ALE bug where entity.selected doesn't get set to false when the creation of an entity is undone (caused problems with SelectedObjects)
SelectedObjects now contains proxied entities instead of the entities themselves

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -2832,7 +2832,18 @@ function addObjBoxResize() {
...
+    let oldCode = m_down.toString();
+    let newCode = oldCode.replace("in newbie.pm) { ldn", `
+        in newbie.pm) {
+        lnd('es[' + newid + '].selected=false;');
+        ldn
+    `);
+    if (oldCode === newCode) { 
+        aleiLog(WARN, "m_down direct code replacement failed (selected fix)");
+    }
+    let og_mdown = eval("(" + newCode + ")");
```
Comment (Molisson) [inline src/main.js:2845]: Hm? Why do we need `eval` here?

**Review by Molisson — COMMENTED** [review]
Reviewed, concerned about `eval` though

File: src/main.js (hunk 1 above)
Comment (LisABC) [reply src/main.js:2845]: Jeje is doing changes on top of function code, and in order to apply changes, you need to get function object, which you can get by evalling.

Comment (Molisson) [reply src/main.js:2845]: Oh, that makes sense

**Comment by Molisson** [comment]
Merging

**Pull request merged**